src/components/canvas/OutlinedObjects.tsx

```diff
@@ -0,0 +1,437 @@
+    const yAxisV = zAxisV.clone().cross(xAxisV).normalize()
+    const rotationMatrix = new THREE.Matrix4().makeBasis(xAxisV, yAxisV, zAxisV)
+
+    return new THREE.Quaternion().setFromRotationMatrix(rotationMatrix)
+  }, [xAxis, zAxis])
+
+  const torusGeometry = React.useMemo(() => {
+    return new THREE.TorusGeometry(radius, circleWidth * scale, 6, 60)
+  }, [radius, scale])
+
+  return (
+    <mesh position={center} quaternion={quaternion} geometry={torusGeometry} material={transparentMaterial} renderOrder={1000} />
+  )
+}
+
+const OutlinedObject: React.FC<{ group: string, id: number }> = ({ children, group, id }) => {
+  const outlinedMeshes = useOutlinesStore(s => s.outlinedMeshes)
+  const setOutlinedMeshes = useOutlinesStore(s => s.setOutlinedMeshes)
+  const removeMesh = useOutlinesStore(s => s.removeMesh)
+
+  const groupRef = React.useRef<THREE.Group>(null!)
+
+  useFrame(() => {
+    if (!outlinedMeshes[group]?.[id] && groupRef.current) {
+      const meshes_: THREE.Object3D[] = []
+  
+      groupRef.current.traverse(o => {
+        if (o.type === 'Mesh') {
+          meshes_.push(o)
+        }
+      })
+  
+      setOutlinedMeshes(group, id, meshes_)
+    }
+  })
+
+  React.useEffect(() => {
+    return () => removeMesh(group, id)
+  }, [])
+
+  return (
+    <group ref={groupRef}>
+      {children}
+    </group>
+  )
+}
+
+const OutlinedProduct: React.FC<{ group: string, id: number }> = ({ group, id }) => {
+  const { scene } = useThree()
+  
+  const outlinedMeshes = useOutlinesStore(s => s.outlinedMeshes)
+  const setOutlinedMeshes = useOutlinesStore(s => s.setOutlinedMeshes)
+  const removeMesh = useOutlinesStore(s => s.removeMesh)
+
+  useFrame(() => {
+    if (!outlinedMeshes[group]?.[id]) {
+      const obj = scene?.getObjectByName(id.toString())
+      const meshes_: THREE.Object3D[] = []
+  
+      obj?.traverse(o => {
+        if (o.type === 'Mesh') {
+          meshes_.push(o)
+        }
+      })
+  
+      setOutlinedMeshes(group, id, meshes_)
+    }
+  })
+
+  React.useEffect(() => {
+    return () => removeMesh(group, id)
+  }, [])
+
+  return null
+}
+
+export function OutlinedObjects({ drawingId, info, group }: { drawingId: DrawingID, info: InteractionInfo, group: string }) {
+  const objClass = useDrawing(drawingId, d => d.structure.tree[info.objectId]?.class)
+
+  if (objClass && !info.graphicId) {
+    // Assembly node
+    if (ccUtils.base.isA(objClass, CCClasses.IProductReference)) {
+      return (
+        <OutlinedProduct key={info.objectId} group={group} id={info.objectId} />
+      )
+    }
+  
+    // Constraint
+    if (ccUtils.base.isA(objClass, CCClasses.CCHLConstraint)) {
+      const mateRefIds = getMateRefIds(drawingId, info.objectId)
+      return (
+        <>
+          {mateRefIds?.map(id => (
+            <OutlinedProduct key={id} group={group} id={id} />
+          )) || null}
+        </>
+      )
+    }
+  
+    // Feature
+    if (objClass === CCClasses.CCWorkPoint) {
```

Comment:
CCClasses should always be checked by `ccUtils.base.isA`

Reply:
Replaced all the direct checks with it in [a6e29f6](https://github.com/awv-informatik/buerligons/pull/6/commits/a6e29f67ff5161748ed5110a3a6ec7b194555f03)